src/runtime_src/core/tools/common/ReportHost.cpp

```diff
@@ -108,6 +108,10 @@ ReportHost::writeReport(const xrt_core::device * _pDevice,
 
   _output << "Devices\n";
   boost::property_tree::ptree& available_devices = _pt.get_child("host.devices", empty_ptree);
+
+  if(available_devices.size() == 0)

```

Comment:
Since this code is determining if the collection is empty, it would be better to use:
` if(available_devices.empty())`
it is faster and easier to read.